pallets/restricted-xtokens/src/mock/mod.rs

```diff
@@ -0,0 +1,328 @@
+// GNU General Public License for more details.
+
+#![cfg(test)]
+
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+use sp_io::TestExternalities;
+use sp_runtime::{traits::Zero, AccountId32, BoundedVec};
+use xcm_executor::{traits::WeightTrader, Assets};
+use xcm_simulator::{decl_test_network, decl_test_parachain, decl_test_relay_chain, TestExt};
+
+use super::*;
+use crate as restricted_xtokens;
+
+pub mod para;
+pub mod relay;
+
+pub const RESTRICTED_SENDER: AccountId32 = AccountId32::new([0u8; 32]);
+pub const BOB: AccountId32 = AccountId32::new([1u8; 32]);
+pub const RESTRICTED_RECEIVER: AccountId32 = AccountId32::new([2u8; 32]);
+
+fn para_a_rreceiver_relay() -> MultiLocation {
+	MultiLocation::new(
+		1,
+		X1(Junction::AccountId32 {
+			network: None,
+			id: RESTRICTED_RECEIVER.into(),
+		}),
+	)
+}
+
+fn para_a_rreceiver_para_a() -> MultiLocation {
+	MultiLocation::new(
+		0,
+		X1(Junction::AccountId32 {
+			network: None,
+			id: RESTRICTED_RECEIVER.into(),
+		}),
+	)
+}
+
+fn para_a_rreceiver_para_b() -> MultiLocation {
+	MultiLocation::new(
+		1,
+		X2(
+			Junction::Parachain(2),
+			Junction::AccountId32 {
+				network: None,
+				id: RESTRICTED_RECEIVER.into(),
+			},
+		),
+	)
+}
+
+#[derive(
+	Encode,
+	Decode,
+	Eq,
+	PartialEq,
+	Copy,
+	Clone,
+	RuntimeDebug,
+	PartialOrd,
+	Ord,
+	codec::MaxEncodedLen,
+	TypeInfo,
+)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum CurrencyId {
+	/// Relay chain token.
+	R,
+	/// Parachain A token.
+	A,
+	/// Parachain A A1 token.
+	A1,
+	/// Parachain B token.
+	B,
+	/// Parachain B B1 token
+	B1,
+}
+
+pub struct CurrencyIdConvert;
+impl sp_runtime::traits::Convert<CurrencyId, Option<MultiLocation>> for CurrencyIdConvert {
+	fn convert(id: CurrencyId) -> Option<MultiLocation> {
+		match id {
+			CurrencyId::R => Some(Parent.into()),
+			CurrencyId::A => Some(
+				(
+					Parent,
+					Parachain(1),
+					Junction::from(BoundedVec::try_from(b"A".to_vec()).unwrap()),
+				)
+					.into(),
+			),
+			CurrencyId::A1 => Some(
+				(
+					Parent,
+					Parachain(1),
+					Junction::from(BoundedVec::try_from(b"A1".to_vec()).unwrap()),
+				)
+					.into(),
+			),
+			CurrencyId::B => Some(
+				(
+					Parent,
+					Parachain(2),
+					Junction::from(BoundedVec::try_from(b"B".to_vec()).unwrap()),
+				)
+					.into(),
+			),
+			CurrencyId::B1 => Some(
+				(
+					Parent,
+					Parachain(2),
+					Junction::from(BoundedVec::try_from(b"B1".to_vec()).unwrap()),
+				)
+					.into(),
+			),
+		}
+	}
+}
+impl sp_runtime::traits::Convert<MultiLocation, Option<CurrencyId>> for CurrencyIdConvert {
+	fn convert(l: MultiLocation) -> Option<CurrencyId> {
+		let mut a: Vec<u8> = "A".into();
+		a.resize(32, 0);
+		let mut a1: Vec<u8> = "A1".into();
+		a1.resize(32, 0);
+		let mut b: Vec<u8> = "B".into();
+		b.resize(32, 0);
+		let mut b1: Vec<u8> = "B1".into();
+		b1.resize(32, 0);
+		if l == MultiLocation::parent() {
+			return Some(CurrencyId::R);
+		}
```

Comment:
I think this is unnecessarily clutter-y. If we would use a `GeneralIndex` (`u128`) that we map for each currency, then it's a lot simpler and we could even define that map at the enum definition level (i.e, `pub enum CurrencyId { R = 0, A = 1, etc`.)

Reply:
I copy pasta that one from acala. While I agree I think for the given 5 enum variants this logic is fine and I do not expect to change it often in the futre. Wdyt?